examples/cookie_extractor.rs

```diff
@@ -0,0 +1,16 @@
+#![feature(async_await, futures_api)]
+
+use tide::Cookies;
+
+async fn hello_cookies(cookies: Cookies) -> String {
```

Comment:
Some documentation explaining what the `Cookies` type is could be helpful. In the long run, this could be an example of how the `Extract` trait works in a guide-like doc.

Reply:
@blgBV you mean to say that Cookies type is a wrapper around CookieJar and open for future extensions? something in those lines? or more documentation on the example or both?
I am happy to add more documentation but need some pointers

Reply:
@mmrath I can't speak for @bIgBV but I was thinking explaining high-level what this function does, and why it works this way might be useful. Something like:

> Extract the cookies from the HTTP header into a Map. This is done through the `Cookies` extractor.

To people unfamiliar with Tide this method might seem rather magical, and having a few pointers on where to look might be useful.

Separate from this we should probably write some docs on how extractors work, in addition to the [existing blog post](https://rust-lang-nursery.github.io/wg-net/2018/10/16/tide-routing.html).

Reply:
Yup, this is basically what I had in mind.